src/webui/api/torrentscontroller.cpp

```diff
@@ -1129,17 +1129,23 @@ void TorrentsController::editTrackerAction()
     const auto id = BitTorrent::TorrentID::fromString(params()[u"hash"_s]);
     const QString origUrl = params()[u"origUrl"_s];
     const QString newUrl = params()[u"newUrl"_s];
+    // min tier is 0, so -1 indicates to keep tier unchanged
+    const int newTier = parseInt(params()[u"tier"_s]).value_or(-1);
 
     BitTorrent::Torrent *const torrent = BitTorrent::Session::instance()->getTorrent(id);
     if (!torrent)
         throw APIError(APIErrorType::NotFound);
 
     const QUrl origTrackerUrl {origUrl};
     const QUrl newTrackerUrl {newUrl};
-    if (origTrackerUrl == newTrackerUrl)
+    const bool isTrackerUrlChanged = origTrackerUrl != newTrackerUrl;
+    const bool isNewTierSpecified = newTier >= 0;
+    if (!isTrackerUrlChanged && !isNewTierSpecified)
         return;
```

Comment:
This endpoint has logic to return early if the URL is unchanged. With the addition of the `tier` param, some effort was required to maintain this functionality.

The logic here still returns early if the URL is unchanged **and** the tier is unspecified. If the tier is specified, the logic for returning early is moved to the loop, where it checks that both the URL and tier are unchanged before returning early.